src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,241 @@
+    super(applicationParameters, engine)
+
+    const { applicativeHostUrl, requestMethod, authentication, csvDelimiter, mapping, proxy = null } = applicationParameters.CsvToHttp
+
+    // Creation of an entity request with is composed of all necessaries informations
+    this.request = {}
+    this.request.host = applicativeHostUrl
+    this.request.method = requestMethod
+    this.request.authentication = authentication
+
+    this.mapping = mapping
+    this.mapping.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.httpBody = []
+    this.canHandleFiles = true
+  }
+
+  /**
+   * Send the file.
+   * @param {String} filePath - The path of the file
+   * @return {Promise} - The send status
+   */
+  async handleFile(filePath) {
+    // Verify that the file receive is a csv one
+    const regexExp = RegExp('.csv$')
+
+    if (!regexExp.test(filePath)) {
+      this.logger.error(`Invalid file format, file (${filePath} skipped`)
+      return ApiHandler.STATUS.LOGIC_ERROR
+    }
+
+    await this.parseAndMapp(filePath)
+
+    if (this.httpBody !== []) {
+      await this.sendData()
+      this.logger.silly('Request send successfully')
+    }
+    return ApiHandler.STATUS.SUCCESS
+  }
+
+  /**
+   * @param {string} filePath - PAth of the file to parse
+   * @return {Promise} - Promise
+   */
+  parseAndMapp(filePath) {
+    return new Promise((resolve, reject) => {
+      try {
+        // Initialize array wich will be filled at each step
+        const csvDataParsed = []
+        const csvFile = fs.createReadStream(filePath)
+        // Parse the csv file into a json object
+        Papa.parse(csvFile, {
+          delimiter: this.mapping.csvDelimiter,
+          header: true,
+          dynamicTyping: true,
+          error: (error) => {
+            this.logger.error(error)
+            return ApiHandler.STATUS.LOGIC_ERROR
+          },
+          step: (step) => {
+            csvDataParsed.push(step.data)
+          },
+          complete: () => {
+            // The csv parsing is a success so we begin to map it
+            // Initialize the body to an empty array
+            this.httpBody = []
+            this.convertToHttpBody(csvDataParsed)
+          },
+        })
+        resolve()
+      } catch (error) {
+        this.logger.error(error)
+        reject()
+      }
+    })
+  }
+
+  /**
+   * @param {Object} jsonObject - jsonObj
+   * @return {Promise} - The send status
+   */
+  convertToHttpBody(jsonObject) {
+    return new Promise((resolve, reject) => {
+      try {
+        if (jsonObject.length > 0) {
+          jsonObject.forEach((jsonObj) => {
+            const response = this.convertCSVRowIntoHttpBody(jsonObj)
+            // Test the result of the mapping/convertion before add it in the httpBody
+            if (response) {
+              this.httpBody.push(response)
+            }
+          })
+        }
+        this.logger.silly('End of mapping (csv file to http body)')
```

Comment:
this log does not seem useful to me : in case of success, the final log is used, if not an error log is used